addon/request-assertion.js

```diff
@@ -0,0 +1,90 @@
+/* * RequestAssertion implements a simple DSL for asserting that specific requests were made.
```

Comment:
Not sure if `RequestAssertion` is the best name since 'assertion' is a loaded term. Maybe something like `RequestChecker`?